lib/routes/routeBackbeat.js

```diff
@@ -407,24 +430,46 @@ function completeMultipartUpload(request, response, log, callback) {
         return callback(err);
     }
     const storageLocation = request.headers['x-scal-storage-class'];
+    const sourceVersionId = request.headers['x-scal-version-id'];
     const uploadId = request.headers['x-scal-upload-id'];
+    const userMetadata = request.headers['x-scal-user-metadata'];
+    const contentType = request.headers['x-scal-content-type'];
+    const cacheControl = request.headers['x-scal-cache-control'];
+    const contentDisposition = request.headers['x-scal-content-disposition'];
+    const contentEncoding = request.headers['x-scal-content-encoding'];
     const data = [];
     let totalLength = 0;
     request.on('data', chunk => {
         totalLength += chunk.length;
         data.push(chunk);
     });
     request.on('end', () => {
-        let Part;
+        let parts;
         try {
-            Part = JSON.parse(Buffer.concat(data), totalLength);
+            parts = JSON.parse(Buffer.concat(data), totalLength);
         } catch (e) {
             // FIXME: add error type MalformedJSON
             return callback(errors.MalformedPOSTRequest);
         }
+        const partList =
+            getPartList(parts, request.objectKey, uploadId, storageLocation);
+        // Azure client will set user metadata at this point.
+        const metaHeaders = {
+            'scal-replication-status': 'REPLICA',
+            'scal-version-id': sourceVersionId,
+        };
+        if (userMetadata) {
+            Object.assign(metaHeaders, JSON.parse(userMetadata));
```

Comment:
I think you should not trust `userMetadata` since it comes directly from the received headers, i.e. put the `JSON.parse` in a `try..catch` and return a request error on failure. You may add additional checks if need be but I don't know if there's any useful check.

Reply:
Nice catch. Updated.